src/containers/Article.js

```diff
@@ -54,32 +65,115 @@ class Article extends PureComponent {
+    this.handleScroll = _.debounce(this._handleScroll.bind(this), 500)
+    this.handleVisibilityChange = _.throttle(
+      this._handleVisibilityChange.bind(this),
+      1000
+    )
+    this.handleUserActivity = _.debounce(
+      this._handleUserActivity.bind(this),
+      500
+    )
   }
 
-  startTimer() {
-    if (this.timerId.current) {
-      window.clearTimeout(this.timerId.current)
+  startReadingCountTimer() {
+    if (this.readingCountTimerId.current) {
+      window.clearTimeout(this.readingCountTimerId.current)
     }
-    this.timerId.current = window.setTimeout(() => {
+    this.readingCountTimerId.current = window.setTimeout(() => {
       this.setState({ isReachedArticleReadTargetTime: true })
     }, articleReadCountConditionConfig.reading_time)
   }
 
-  componentDidMount() {
-    const { fontLevel, changeFontLevel, slugToFetch } = this.props
-    window.addEventListener('scroll', this.handleScroll)
+  calculateActiveTime() {
+    const elapsedTime = Math.floor(Date.now() - this.state.startReadingTime)
+    this.setState(prevState => ({
+      activeTime: prevState.activeTime + elapsedTime,
+      startReadingTime: Date.now(),
+    }))
+  }
 
-    // Fetch the full post
-    this.fetchAFullPostWithCatch(slugToFetch)
-    // Start timer if post is fetched from SSR
-    this.startTimer()
+  startInactiveTimer() {
+    if (this.inactiveTimerId.current) {
+      window.clearTimeout(this.inactiveTimerId.current)
+    }
+    this.inactiveTimerId.current = window.setTimeout(() => {
+      this.calculateActiveTime()
+      this.setState({
+        isActive: false,
+      })
+    }, articleReadTimeConditionConfig.inactive_time)
+  }
+
+  sendReadCount() {
+    const { jwt, userID, postID, setUserAnalyticsData } = this.props
+    setUserAnalyticsData(jwt, userID, postID, { readPostCount: true })
+  }
+
+  sendActiveTime() {
+    const { jwt, userID, postID, setUserAnalyticsData } = this.props
+    const { activeTime } = this.state
+    const activeSec = Math.round(activeTime / 1000)
+    setUserAnalyticsData(jwt, userID, postID, { readPostSec: activeSec })
+    this.setState({
+      activeTime: 0,
+    })
+  }
+
+  _handleVisibilityChange() {
+    if (document.visibilityState === 'hidden') {
+      if (
+        this.state.activeTime >= articleReadTimeConditionConfig.min_active_time
+      ) {
+        this.sendActiveTime()
+        this.setState({
+          isActive: false,
+        })
+      }
+    }
+  }
+
+  _handleUserActivity() {
+    if (!this.state.isActive) {
+      this.setState({
+        isActive: true,
+        startReadingTime: Date.now(),
+      })
+    } else {
+      this.startInactiveTimer()
+    }
+  }
+
+  componentDidMount() {
+    const { fontLevel, changeFontLevel, slugToFetch, isAuthed } = this.props
+
+    if (isAuthed) {
+      window.addEventListener('scroll', this.handleScroll)
+      // Fetch the full post
+      this.fetchAFullPostWithCatch(slugToFetch)
```

Comment:
Question: why do we move `fetchAFullPostWithCatch` inside `isAuthed` rule ?

Reply:
fixed